release.py

```diff
@@ -0,0 +1,212 @@
+    verify_and_warn()
+
+    # Compute the new version number
+    old_version = get_version('orchestra')
+    new_version = increment_version_number(old_version, args.version_part)
+    print('Updating Orchestra version from {} to {}'
+          .format(old_version, new_version))
+
+    # Update orchestra/__init__.py with the new version
+    update_version('orchestra', new_version, fake=args.fake)
+
+    # Commit the update to the repo
+    commit_and_push(fake=args.fake)
+
+    # Create a new tag for the release
+    tag_str = tag_release(new_version, fake=args.fake)
+
+    # Release the new version on pypi
+    pypi_release(tag_str, fake=args.fake)
+
+
+def verify_and_warn():
+    # verify that we're on the master branch
+    branch_cmd = ['git', 'symbolic-ref', '--short', 'HEAD']
+    cur_branch = check_output(branch_cmd).decode().strip().lower()
+    if cur_branch != 'master':
+        print('This script must be run from the master branch. Exiting.')
+        exit()
+
+    # verify that master is up to date
+    rev_cmd = ['git', 'rev-list']
+    unpushed_cmd = rev_cmd + ['origin/master..']
+    unpushed_commits = check_output(unpushed_cmd).decode().strip()
+    if unpushed_commits:
+        print('Your branch has commits not yet on origin/master. Exiting.')
+        exit()
+
+    unpulled_cmd = rev_cmd + ['..origin/master']
+    unpulled_commits = check_output(unpulled_cmd).decode().strip()
+    if unpulled_commits:
+        print('Your branch is behind origin/master. Exiting.')
+        exit()
+
+    # verify that there are no outstanding changes
+    status_cmd = ['git', 'status', '--porcelain']
+    status_output = check_output(status_cmd).decode().strip()
+    if status_output:
+        print('There are outstanding changes to your branch. Exiting.')
+        exit()
+
+    print('WARNING: running this script will increment the Orchestra version. '
+          'This involves pushing a new commit to master and releasing a new '
+          'PyPI distribution, changes you CANNOT TAKE BACK. Before '
+          'proceeding, ensure that you are on the master branch, have pulled '
+          'the latest changes, and have no outstanding local changes.')
+    ack = input('ARE YOU SURE YOU WISH TO PROCEED? (y/n): ').lower()
+    while ack not in ['y', 'n']:
+        ack = input('Please respond with "y" or "n": ').lower()
+    if ack == 'n':
+        exit()
+
+
+def increment_version_number(old_version, version_part):
+    version_parsed = StrictVersion(old_version)
+    major, minor, patch = version_parsed.version
+    if version_part == 'major':
+        major += 1
+        minor = 0
+        patch = 0
+    elif version_part == 'minor':
+        minor += 1
+        patch = 0
+    elif version_part == 'patch':
+        patch += 1
+    new_version = '{}.{}.{}'.format(major, minor, patch)
+    return new_version
+
+
+def version_file(package):
+    return os.path.join(package, '__init__.py')
+
+
+def get_version(package):
```

Comment:
quick comment explaining that this makes it possible to live outside of django, and that it can be used in setup.py, which can't import what we're installing